core/src/main/java/com/wttech/aem/acm/core/assist/resource/ResourceScanner.java

```diff
@@ -11,29 +11,43 @@
 public class ResourceScanner {
 
     /**
-     * Suggest the node and children if path is having trailing slash, examples:
+     * Suggest the node and children examples:
      * <p>
      * / => root node suggestion (with node type etc), and its children /apps, /content, etc (exception)
      * /content => only content node suggestion
      * /content/ => only content node children
      * /content/mysite => only mysite node suggestion
      * /content/mysite/ => only mysite node children
+     * /content/mys => suggestion on /content/ children whose names start with "mys"
      * etc.
      */
     public Stream<Resource> forPattern(ResourceResolver resolver, String pattern) {
         if (StringUtils.isBlank(pattern)) {
             return Stream.empty();
         }
 
-        Resource resource = resolver.getResource(pattern);
-        if (resource == null) {
-            return Stream.empty();
-        }
-
         if (pattern.endsWith("/")) {
-            return StreamSupport.stream(resource.getChildren().spliterator(), false);
+            return getChildren(resolver, pattern);
         } else {
-            return Stream.of(resource);
+            int lastSlashIndex = pattern.lastIndexOf('/');
+            if (lastSlashIndex == -1) {
+                Resource resource = resolver.getResource(pattern);
+                return resource == null ? Stream.empty() : Stream.of(resource);
+            }
+            String directory = pattern.substring(0, lastSlashIndex);
+            String filename = pattern.substring(lastSlashIndex + 1);
+            Stream<Resource> children = getChildren(resolver, directory);
```

Comment:
do not use getter-like `get` prefix for things which are not getters;

getter should only return reference; more fits here e.g `listChildren`